### PR TITLE
Article administration

### PR DESCRIPTION
**Pull request by adrianvie**
please approve

### CODE REVIEW

File: src/components/layouts/articleAdministration/ProductOverview.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+  const openNewProductDialog = () => {
+    setNewProductDialogViewState(true);
+  }
+
+  const openNewNpkProductDialog = () => {
+    setNewNpkProductDialogViewState(true);
+  }
+
+/*  const submitNewArticle = (newArticleData) => {
```
Comment (RicardoMonteiroSimoes) [inline src/components/layouts/articleAdministration/ProductOverview.js]: Remove stuff like this if it's not used

File: src/components/layouts/articleAdministration/ProductOverview.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      <AddProductDialog
+          show={showNewProductDialog}
+          articles={productData}
+          onCancel={closeNewProductDialog}
+          onSubmit={loadNewProducts}
+          products={products}
+          npks={npks}
+          setNpks={setNpks}
+      />
+
```
Comment (RicardoMonteiroSimoes) [inline src/components/layouts/articleAdministration/ProductOverview.js]: these 2 dialogs appear to do the same? Or why do both take the same parameters? If my understanding is right, one creates articles without NPK (but still has npks as parameter?).

File: src/components/layouts/articleAdministration/addNpkProductDialog.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+  const parseArticleData = (articleData) => {
+    articleData.price = parseInt(articleData.price);
+    articleData.name = (articleData.npk.split(" "))[1]
+    articleData.npk = (articleData.npk.split(" "))[0]
+    saveNewArticle(articleData);
+  }
+
+  useEffect(() => {
+    getNpks(setErrorMessage, setNpks)
```
Comment (RicardoMonteiroSimoes) [inline src/components/layouts/articleAdministration/addNpkProductDialog.js]: Are the npks needed outside of this dialog? If not, then the state for the npks shouldn't be outisde of the dialog, but within it.

File: src/components/layouts/articleAdministration/productTable.js [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+import Switch from '@material-ui/core/Switch';
+import DeleteIcon from '@material-ui/icons/Delete';
+import FilterListIcon from '@material-ui/icons/FilterList';
+import {getProducts} from "../../connectionHandler/connectionHandler";
+
+
+const ProductTable = ({setErrorMessage, products, setProducts}) => {
+
+  useEffect(() => {
+    getProducts(setErrorMessage, setProducts)
```
Comment (RicardoMonteiroSimoes) [inline src/components/layouts/articleAdministration/productTable.js]: Here the same, if products arent needed outside this dialog, use the products state inside of this class

**Review by RicardoMonteiroSimoes — CHANGES_REQUESTED** [review]

File: src/components/layouts/articleAdministration/ProductOverview.js (hunk 1 above)
Comment (adrianvie) [reply src/components/layouts/articleAdministration/ProductOverview.js]: thought I would still need it

removed it anyway

File: src/components/layouts/articleAdministration/ProductOverview.js (hunk 2 above)
Comment (adrianvie) [reply src/components/layouts/articleAdministration/ProductOverview.js]: your right, they are gone

File: src/components/layouts/articleAdministration/addNpkProductDialog.js (hunk 3 above)
Comment (adrianvie) [reply src/components/layouts/articleAdministration/addNpkProductDialog.js]: moved it out into the upper component

File: src/components/layouts/articleAdministration/productTable.js (hunk 4 above)
Comment (adrianvie) [reply src/components/layouts/articleAdministration/productTable.js]: moved the effect out of this class into the upper class

**Review by RicardoMonteiroSimoes — APPROVED** [review]

**Pull request merged**